File: docs/public/specs/python/density1d.py
import vgplot as vg

flights = vg.parquet("data/flights-200k.parquet")

brush = vg.selection.crossfilter()
bandwidth = vg.param(20)

view = vg.vconcat(
    vg.slider(label="Bandwidth (σ)", bind=bandwidth, min=0.1, max=100, step=0.1),
    vg.plot(
        vg.density_y(
            data="flights",
            filter_by=brush,
            x="delay",
            fill="#888",
            fill_opacity=0.5,
            bandwidth=bandwidth,
        ),
        vg.interval_x(bind=brush),
        vg.y_axis(None),
        vg.x_domain("Fixed"),
        vg.width(600),
        vg.margin_left(10),
        vg.height(200),
    ),
    vg.plot(
        vg.density_y(
            data="flights",
            filter_by=brush,
            x="distance",
            fill="#888",
            fill_opacity=0.5,
            bandwidth=bandwidth,
        ),
        vg.interval_x(bind=brush),
        vg.y_axis(None),
        vg.x_scale("log"),
        vg.x_domain("Fixed"),
        vg.width(600),
        vg.margin_left(10),
        vg.height(200),
    ),
)

view

File: packages/vgplot/vgplot/test/specs/density1d.js
import { dataPath } from '../util/data-path.js';

export default async function(vg) {
  await vg.context.coordinator.exec([
    vg.loadParquet("flights", dataPath("flights-200k.parquet"))
  ]);

  const $brush = vg.Selection.crossfilter();
  const $bandwidth = vg.Param.value(20);

  return vg.vconcat(
    vg.slider({label: "Bandwidth (σ)", as: $bandwidth, min: 0.1, max: 100, step: 0.1}),
    vg.plot(
      vg.densityY(
        vg.from("flights", {filterBy: $brush}),
        {x: "delay", fill: "#888", fillOpacity: 0.5, bandwidth: $bandwidth}
      ),
      vg.intervalX({as: $brush}),
      vg.yAxis(null),
      vg.xDomain(vg.Fixed),
      vg.width(600),
      vg.marginLeft(10),
      vg.height(200)
    ),
    vg.plot(
      vg.densityY(
        vg.from("flights", {filterBy: $brush}),
        {x: "distance", fill: "#888", fillOpacity: 0.5, bandwidth: $bandwidth}
      ),
      vg.intervalX({as: $brush}),
      vg.yAxis(null),
      vg.xScale("log"),
      vg.xDomain(vg.Fixed),
      vg.width(600),
      vg.marginLeft(10),
      vg.height(200)
    )
  );
}
